### PR TITLE
updated pg flex workbooks with numerous bug fixes and cosmetic changes

### PR DESCRIPTION
**Pull request by andfrazier**
Cosmetic and bug fixes

### CODE REVIEW

File: Workbooks/PostgreSQL Flexible Server/Performance/Autovacuum Blockers/Autovacuum Blockers.workbook [hunk 1]
```diff
@@ -1206,7 +1206,7 @@
                 {
                   "type": 1,
                   "content": {
-                    "json": "- If the transaction duration is > {MIN_XACT_TIME_MS_THRESHOLD_FOR_LONG_XACTS} ms, we consider it as long-running transaction.\r\n- Prevents Autovacuum from reclaiming space in the affected tables\r\n- Can cause bloat in tables and indexes, leading to degraded performance over time\r\n- Best practice is to keep transactions as short as possible",
+                    "json": "- Transactions with a duration exceeding 7,200,000 ms are classified as long-running transactions.\r\n- Long-running transactions do hinder Autovacuum's ability to reclaim space in affected tables.\r\n- These transactions may contribute to table and index bloat, resulting in performance degradation over time.\r\n- To maintain optimal performance, it is recommended to keep transactions as brief as possible.",
```
Comment (seernu) [inline Workbooks/PostgreSQL Flexible Server/Performance/Autovacuum Blockers/Autovacuum Blockers.workbook:1209]: Can you not hardcode 7,200,000 ms. Please fallback to MIN_XACT_TIME_MS_THRESHOLD_FOR_LONG_XACTS

File: Workbooks/PostgreSQL Flexible Server/Performance/Autovacuum Blockers/Autovacuum Blockers.workbook [hunk 2]
```diff
@@ -1413,7 +1413,7 @@
...
+                          "json": "#### Stats on long running transactions\r\n\r\n**Note: In this troubleshooting guide, we classify transactions with a duration of more than 7,200,000 milliseconds as long-running transactions.**\r\n\r\nConsideration:\r\n\r\n* Connection Duration **(blue)** - Duration of session or connection (collection_time-backend_start). Higher values may be observed if you are using a connection pool.\r\n\r\n* Transaction Duration **(green)** - Duration of transaction (collection_time-xact_start)\r\n\r\nLong-running transactions are based on the pg_stat_activity view in PostgreSQL. To learn more about these columns, please visit the PostgreSQL documentation at https://www.postgresql.org/docs/current/monitoring-stats.html#MONITORING-PG-STAT-ACTIVITY-VIEW.",
```
Comment (seernu) [inline Workbooks/PostgreSQL Flexible Server/Performance/Autovacuum Blockers/Autovacuum Blockers.workbook:1416]: again do not hardcode the value of MIN_XACT_TIME_MS_THRESHOLD_FOR_LONG_XACTS

File: Workbooks/PostgreSQL Flexible Server/Performance/High IOPS Usage/High IOPS Usage.workbook [hunk 3]
```diff
@@ -655,7 +647,7 @@
     {
       "type": 1,
       "content": {
-        "json": "### Query Store is currently disabled\r\nThis troubleshooting guide depends on Query Store data. You need to enable it by setting the dynamic server parameter `pg_qs.query_capture_mode` to either ALL or TOP, and configure logging settings for the `PostgreSQL Query Store Runtime` category to send the data to a log analytics workspace, please refer to this guide for detailed instructions: https://learn.microsoft.com/azure/postgresql/flexible-server/howto-configure-and-access-logs.",
+        "json": "### Query Store is currently disabled.\r\nQuery Performance Insight depends on Query Store data. You need to enable it by setting the dynamic server parameter `pg_qs.query_capture_mode` to either ALL or TOP. Refer to this guide for detailed instructions: https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-query-store",
```
Comment (seernu) [inline Workbooks/PostgreSQL Flexible Server/Performance/High IOPS Usage/High IOPS Usage.workbook:650]: This is Troubleshooting guide and not QPI

File: Workbooks/PostgreSQL Flexible Server/Performance/High IOPS Usage/High IOPS Usage.workbook [hunk 4]
```diff
@@ -726,7 +718,7 @@
     {
       "type": 1,
       "content": {
-        "json": "### Query Store Wait Sampling is currently disabled\r\nThis troubleshooting guide depends on Query Store wait sampling data. You need to enable it by setting the dynamic server parameter `pgms_wait_sampling.query_capture_mode` to All, and configuring the Diagnostic Settings destination to log analytics. Please refer to the following guide for detailed instructions: https://learn.microsoft.com/azure/postgresql/flexible-server/howto-configure-and-access-logs.",
+        "json": "### Query Store Wait Sampling is currently disabled\r\nQuery Performance Insight depends on Query Store wait sampling data. You need to enable it by setting the dynamic server parameter `pgms_wait_sampling.query_capture_mode` to All. Refer to this guide for detailed instructions: https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-query-store",
```
Comment (seernu) [inline Workbooks/PostgreSQL Flexible Server/Performance/High IOPS Usage/High IOPS Usage.workbook:721]: Troubleshooting guide and not QPI

File: Workbooks/PostgreSQL Flexible Server/Performance/High Memory Usage/High Memory Usage.workbook [hunk 5]
```diff
@@ -540,7 +539,7 @@
           {
             "type": 1,
             "content": {
-              "json": "### Query Store is currently disabled\r\nThis troubleshooting guide depends on Query Store data. You need to enable it by setting the dynamic server parameter `pg_qs.query_capture_mode` to either ALL or TOP, and configure logging settings for the `PostgreSQL Query Store Runtime` category to send the data to a log analytics workspace, please refer to this guide for detailed instructions: https://learn.microsoft.com/azure/postgresql/flexible-server/howto-configure-and-access-logs.",
+              "json": "### Query Store is currently disabled\r\nQuery Performance Insight depends on Query Store data. You need to enable it by setting the dynamic server parameter `pg_qs.query_capture_mode` to either ALL or TOP. Refer to this guide for detailed instructions: https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-query-store",
```
Comment (seernu) [inline Workbooks/PostgreSQL Flexible Server/Performance/High Memory Usage/High Memory Usage.workbook:542]: It is troubleshooting guide, not QPI

File: Workbooks/PostgreSQL Flexible Server/Performance/High Memory Usage/High Memory Usage.workbook [hunk 6]
```diff
@@ -1428,7 +1426,7 @@
...
               "size": 0,
-              "title": "Full history of the PID for the selected TimeRange",
+              "title": "Full History of the PID for the Selected TimeRange",
```
Comment (seernu) [inline Workbooks/PostgreSQL Flexible Server/Performance/High Memory Usage/High Memory Usage.workbook:1429]: Are you sure CamelCase is better?

File: Workbooks/PostgreSQL Flexible Server/Performance/High Memory Usage/High Memory Usage.workbook [hunk 7]
```diff
@@ -1689,7 +1691,7 @@
...
                     "size": 0,
                     "aggregation": 2,
-                    "title": "Top {TopTimeConsumingQueries} queries by Data Usage. Chart is plotted at a granularity of {ChartGrain}",
+                    "title": "Top {TopTimeConsumingQueries} Queries by Data Usage. Chart is plotted at a granularity of {ChartGrain}",
```
Comment (seernu) [inline Workbooks/PostgreSQL Flexible Server/Performance/High Memory Usage/High Memory Usage.workbook:1694]: in most of the workbooks is being used the syntax "Data is plotted" instead of "Chart is plotted". Can you change here as well for consistency.

File: Workbooks/PostgreSQL Flexible Server/Performance/High Temporary Files/High Temporary Files.workbook [hunk 8]
```diff
@@ -2324,12 +2382,12 @@
...
         "size": 4,
-        "title": "QueryId {QueryIdsByTemp} summary in selected TimeRange",
+        "title": "QueryId {QueryIdsByTemp} Summary in Selected TimeRange",
         "showExportToExcel": true,
         "queryType": 0,
-        "resourceType": "microsoft.dbforpostgresql/flexibleservers",
+        "resourceType": "microsoft.resources/resourcegroups",
```
Comment (seernu) [inline Workbooks/PostgreSQL Flexible Server/Performance/High Temporary Files/High Temporary Files.workbook:2390]: why this change? Wount be this a regression?

File: Workbooks/PostgreSQL Flexible Server/Performance/High Temporary Files/High Temporary Files.workbook [hunk 9]
```diff
@@ -2649,29 +2727,25 @@
...
         "size": 0,
-        "title": "QueryID {QueryIdsByTemp} runtime statistics split by time buckets of {ChartGrain} each",
+        "title": "QueryId {QueryIdsByTemp} Runtime Statistics Split by Time Buckets of {ChartGrain} Each",
         "noDataMessageStyle": 4,
         "showExportToExcel": true,
         "queryType": 0,
-        "resourceType": "microsoft.dbforpostgresql/flexibleservers",
+        "resourceType": "microsoft.resources/resourcegroups",
```
Comment (seernu) [inline Workbooks/PostgreSQL Flexible Server/Performance/High Temporary Files/High Temporary Files.workbook:2742]: Why you selected the RG? 

Can you double check if this is not introducing a regression?

File: Workbooks/PostgreSQL Flexible Server/Performance/Query Performance Insights/Query Performance Insights.workbook [hunk 10]
```diff
@@ -1667,7 +1735,7 @@
...
         "size": 0,
         "aggregation": 2,
-        "title": "Top {TopTimeConsumingQueries} queries by Data Usage. Chart is plotted at a granularity of {ChartGrain}",
+        "title": "Top {TopTimeConsumingQueries} Queries by Data-Usage. Chart is plotted at a granularity of {ChartGrain}",
```
Comment (seernu) [inline Workbooks/PostgreSQL Flexible Server/Performance/Query Performance Insights/Query Performance Insights.workbook:1738]: Can you make it consistent and replace "Chart" with "Data"

**Review by seernu — CHANGES_REQUESTED** [review]
Please address the comments before completing the PR, especially the ones where a regression might have been introduced.

File: Workbooks/PostgreSQL Flexible Server/Performance/Autovacuum Blockers/Autovacuum Blockers.workbook (hunk 1 above)
Comment (andfrazier) [reply Workbooks/PostgreSQL Flexible Server/Performance/Autovacuum Blockers/Autovacuum Blockers.workbook:1209]: image

Updated

File: Workbooks/PostgreSQL Flexible Server/Performance/Autovacuum Blockers/Autovacuum Blockers.workbook (hunk 2 above)
Comment (andfrazier) [reply Workbooks/PostgreSQL Flexible Server/Performance/Autovacuum Blockers/Autovacuum Blockers.workbook:1416]: image

Corrected

File: Workbooks/PostgreSQL Flexible Server/Performance/High Memory Usage/High Memory Usage.workbook (hunk 6 above)
Comment (andfrazier) [reply Workbooks/PostgreSQL Flexible Server/Performance/High Memory Usage/High Memory Usage.workbook:1429]: Yes

File: Workbooks/PostgreSQL Flexible Server/Performance/High Temporary Files/High Temporary Files.workbook (hunk 8 above)
Comment (andfrazier) [reply Workbooks/PostgreSQL Flexible Server/Performance/High Temporary Files/High Temporary Files.workbook]: image



Corrected

File: Workbooks/PostgreSQL Flexible Server/Performance/High Temporary Files/High Temporary Files.workbook (hunk 9 above)
Comment (andfrazier) [reply Workbooks/PostgreSQL Flexible Server/Performance/High Temporary Files/High Temporary Files.workbook]: Corrected 


image

**Review by seernu — APPROVED** [review]

**Comment by andfrazier** [comment]
> Cosmetic and bug fixes

## PR Checklist

 
UI fixes for all workbooks. 

 


Query Performance Insight (sample of one tab)
image

 

Troubleshooting guides [TAB High CPU Usage]
image

 

Troubleshooting guides [TAB High Memory Usage]
image

 

Troubleshooting guides [TAB High IOPS Usage]
image

 

Troubleshooting guides [TAB High Temporaty Files]
image

 

Troubleshooting guides [TAB Autovacuum monitoring]
image

 

Troubleshooting guides [TAB Autovacuum Blockers]
image

 


### If adding or updating templates:
* [ ] post a screenshot of templates and/or gallery changes
* [ ] ensure your template has a corresponding gallery entry in the gallery folder
* [ ] If you are adding a new template, add your team and template/gallery file(s) to the CODEOWNERS file. CODEOWNERS entries should be teams, not individuals
* [ ] ensure all steps have meaningful names
* [ ] ensure all parameters and grid columns have display names set so they can be localized
* [ ] ensure that parameters id values are unique __or they will fail PR validation__ (parameter ids are used for localization)
* [ ] ensure that steps names are unique __or they will fail PR validation__ (step names are used for localization)
* [ ] grep `/subscription/` and ensure that your parameters don't have any hardcoded resourceIds __or they will fail PR validation__
* [ ] remove `fallbackResourceIds` and `fromTemplateId` fields from your template workbook __or they will fail PR validation__
